roles/caddy_server/tasks/main.yml

```diff
@@ -1,10 +1,37 @@
 ---
+- name: Import distro-specific variables
+  include_vars: "{{ distrovars }}"
+  when: (distrovars)
+  vars:
+    distrovars: "{{ lookup('first_found', params, errors='ignore') }}"
+    params:
+      skip: true
+      files:
+        - "{{ ansible_distribution | lower }}-{{ ansible_distribution_major_version }}.yml"
+        - "{{ ansible_distribution | lower }}.yml"
+        - "{{ ansible_os_family | lower }}.yml"
+      paths:
+        - 'vars/distro'
+
 - name: Run Checks
-  ansible.builtin.include_tasks: check.yml
+  include_tasks: check.yml
+
+- name: Perform Caddy install
+  include_tasks: install.yml
 
-- name: Perform distro-specific Caddy install
-  ansible.builtin.include_tasks: "install_{{ ansible_os_family | lower }}.yml"
+- name: Caddy is running
```

Comment:
This is a neat way to enable only the desired service while disabling the other one, but it might be a little hard to read - it took me a moment to understand what it does, even with me having written the role. 

Maybe a more descriptive name could help? Or we just split up the task into two separate ones, one for the Caddyfile and one for the json config.